src/test/regress/expected/update_gp.out

```diff
@@ -623,26 +1054,24 @@ insert into t2_13265 values (2, null, 2, 2);
 explain (verbose, costs off)
 update t1_13265 set b = 2 where
 (c, d) not in (select c, d from t2_13265 where a = 2);
-                                                    QUERY PLAN                                                     
--------------------------------------------------------------------------------------------------------------------
+                                                 QUERY PLAN                                                  
+-------------------------------------------------------------------------------------------------------------
  Update on public.t1_13265
-   ->  Explicit Redistribute Motion 3:3  (slice2; segments: 3)
+   ->  Nested Loop Left Anti Semi (Not-In) Join
          Output: t1_13265.a, 2, t1_13265.c, t1_13265.d, t1_13265.ctid, t1_13265.gp_segment_id, t2_13265.ctid
-         ->  Nested Loop Left Anti Semi (Not-In) Join
-               Output: t1_13265.a, 2, t1_13265.c, t1_13265.d, t1_13265.ctid, t1_13265.gp_segment_id, t2_13265.ctid
-               Join Filter: ((t1_13265.c = t2_13265.c) AND (t1_13265.d = t2_13265.d))
-               ->  Seq Scan on public.t1_13265
-                     Output: t1_13265.a, t1_13265.c, t1_13265.d, t1_13265.ctid, t1_13265.gp_segment_id
-               ->  Materialize
+         Join Filter: ((t1_13265.c = t2_13265.c) AND (t1_13265.d = t2_13265.d))
+         ->  Seq Scan on public.t1_13265
+               Output: t1_13265.a, t1_13265.c, t1_13265.d, t1_13265.ctid, t1_13265.gp_segment_id
+         ->  Materialize
+               Output: t2_13265.ctid, t2_13265.c, t2_13265.d, (2)
+               ->  Broadcast Motion 1:3  (slice1; segments: 1)
```

Comment:
Why did this line change?

Reply:
@bandetto

Reply:
This diff without the patch:
```diff
--- a/src/backend/cdb/cdbmutate.c
+++ b/src/backend/cdb/cdbmutate.c
@@ -1237,7 +1237,7 @@ add_slice_to_motion(Motion *motion,
 
                        break;
                case MOTIONTYPE_EXPLICIT:
-                       motion->plan.flow = makeFlow(FLOW_PARTITIONED, numsegments);
+                       motion->plan.flow = makeFlow(FLOW_PARTITIONED, 123);
 
                        /*
                         * TODO: antova - Nov 18, 2010; add a special locus type for

```

Generates the following plan:
```
                                                    QUERY PLAN                                           
          
---------------------------------------------------------------------------------------------------------
----------
 Update on public.t1_13265
   ->  Explicit Redistribute Motion 3:1123  (slice2; segments: 123)
         Output: t1_13265.a, 2, t1_13265.c, t1_13265.d, t1_13265.ctid, t1_13265.gp_segment_id, t2_13265.c
tid
         ->  Nested Loop Left Anti Semi (Not-In) Join
               Output: t1_13265.a, 2, t1_13265.c, t1_13265.d, t1_13265.ctid, t1_13265.gp_segment_id, t2_1
3265.ctid
               Join Filter: ((t1_13265.c = t2_13265.c) AND (t1_13265.d = t2_13265.d))
               ->  Seq Scan on public.t1_13265
                     Output: t1_13265.a, t1_13265.c, t1_13265.d, t1_13265.ctid, t1_13265.gp_segment_id
               ->  Materialize
                     Output: t2_13265.ctid, t2_13265.c, t2_13265.d, (2)
                     ->  Broadcast Motion 3:3  (slice1; segments: 123)
                           Output: t2_13265.ctid, t2_13265.c, t2_13265.d, (2)
                           ->  Seq Scan on public.t2_13265
                                 Output: t2_13265.ctid, t2_13265.c, t2_13265.d, 2
                                 Filter: (t2_13265.a = 2)
 Optimizer: Postgres query optimizer
 Settings: optimizer=off
(17 rows)
```

So I suppose the number is different due to the lack of an Explicit Redistribute Motion.

Reply:
I still don’t understand why Broadcast Motion began to be made from one segment, and not from three?

Reply:
Direct dispatch is disabled in case there is more than one slice:
https://github.com/arenadata/gpdb/blob/5beccaaa4e08a184f2d7c5e4b5d412805ccdce1e/src/backend/cdb/cdbtargeteddispatch.c#L803-L823

Where it is used by EXPLAIN:
https://github.com/arenadata/gpdb/blob/5beccaaa4e08a184f2d7c5e4b5d412805ccdce1e/src/backend/commands/explain.c#L1516-L1530

Therefore when there is no Explicit Redistribute Motion, having only one slice allows for direct dispatch.